fuse_optimizers/test/common_robot_config.yaml

```diff
@@ -0,0 +1,36 @@
+motion_models:
+  unicycle_2d:
+    type: fuse_models::Unicycle2D
+
+sensor_models:
+  unicycle_2d_ignition:
+    type: fuse_models::Unicycle2DIgnition
+    motion_models: ['unicycle_2d']
+  wheel_odometry:
+    type: fuse_models::Odometry2D
+    motion_models: ['unicycle_2d']
+  laser_localization:
+    type: fuse_models::Odometry2D
+    motion_models: ['unicycle_2d']
+
+publishers:
+  odometry_publisher:
+    type: fuse_models::Odometry2DPublisher
+
+
+unicycle_2d:
```

Comment:
I'm curious what you think. Given the optimizer layout above, we could move all of the model-specific configuration into a single place. Like so:
```
motion_models:
  unicycle_2d:
    type: fuse_models::Unicycle2D
    process_noise_diagonal: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]

sensor_models:
  unicycle_2d_ignition:
    type: fuse_models::Unicycle2DIgnition
    motion_models: ['unicycle_2d']
    initial_sigma: [0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
    publish_on_startup: false
  wheel_odometry:
    type: fuse_models::Odometry2D
    motion_models: ['unicycle_2d']
    topic: odom
    differential: true
    twist_target_frame: base_link
  laser_localization:
    type: fuse_models::Odometry2D
    motion_models: ['unicycle_2d']
    topic: pose
    pose_target_frame: map
    twist_target_frame: base_link

publishers:
  odometry_publisher:
    type: fuse_models::Odometry2DPublisher
```

It would require a bit of extra work, as the motion model/sensor/publisher interface would need to change to accept a node handle in the configuration namespace. And that doesn't need to happen in this PR. I'm just curious if you like that better? Or do you want to keep the motion model configuration separate from the optimizer configuration (as is done now)?

Reply:
Uhm, interesting. The config suggestion here would be more compact, but on the other side, I like that with the current configuration we have something like a top view / summary / interface, and the details are **enforced** to be somewhere else.

Certainly, with your suggestion we can still have the model specific parameters later in the file, or in another file, if we want to provide a summary-like config w/o the details (model specific parameters). It's simply not enforced. I believe this is the only different from the config point of view.

Another point would be that with your suggested approach here, things get better organized in the param server, since we have all the config under less namespaces, e.g. `sensor_models` will have all the sensor models params, instead of having each in its own, "top" level namespace.

I certainly don't have a strong opinion. Probably your suggested config format is superior because we can do everything we can with the current one and more, so it's more flexible.

Reply:
Alright. I'll look into what needs to happen to support that. Thanks.

Reply:
Another thing that is still hard to extend, is the `ignition_sensors` parameter, which is a list/array.

It'd be great if we could remove that parameter completely, and add something to the `SensorModel` plugins, so they have a method that indicates whether they're ignition sensor models or not. It sounds like a compile time thing that could be done with traits, but I believe it'd have to be done with a runtime virtual method because we're using plugins.

Thoughts?

Reply:
The parameter could definitely be moved from a separate list to part of the sensor config. Something like:
```
sensor_models:
  unicycle_2d_ignition:
    type: fuse_models::Unicycle2DIgnition
    ignition: true
    motion_models: ['unicycle_2d']
```
And maybe we could think of a better name than "ignition" at some point.

But I view "ignition-ness" more as a situation-specific property and less as a property of the sensor itself. Perhaps someone whats to use a laserscan-to-map match to initialize the state. Or maybe a GPS reading. At least that was the original intention of the system. At this point I'm not sure if reality matches that intention or not. But I will think about it.

Reply:
Makes sense.

I was mostly thinking of the ability of a SensorModel to work as an ignition sensor model being something in the class.

But you're also right in that the user could decide whether to use such a sensor model as a regular one, i.e. not being used for "ignition", or not.

It's probably a good idea to have both things, so if someone sets `ignition: true` on a sensor that doesn't have igniton capabilities, an exception is thrown.